benchmark/single-source/KeyPathPerformanceTests.swift

```diff
@@ -0,0 +1,1668 @@
+  return FixedSizeArray100<ElementType>(
+    element0: 0,
+    element1: 1,
+    element2: 2,
+    element3: 3,
+    element4: 4,
+    element5: 5,
+    element6: 6,
+    element7: 7,
+    element8: 8,
+    element9: 9,
+    element10: 10,
+    element11: 11,
+    element12: 12,
+    element13: 13,
+    element14: 14,
+    element15: 15,
+    element16: 16,
+    element17: 17,
+    element18: 18,
+    element19: 19,
+    element20: 20,
+    element21: 21,
+    element22: 22,
+    element23: 23,
+    element24: 24,
+    element25: 25,
+    element26: 26,
+    element27: 27,
+    element28: 28,
+    element29: 29,
+    element30: 30,
+    element31: 31,
+    element32: 32,
+    element33: 33,
+    element34: 34,
+    element35: 35,
+    element36: 36,
+    element37: 37,
+    element38: 38,
+    element39: 39,
+    element40: 40,
+    element41: 41,
+    element42: 42,
+    element43: 43,
+    element44: 44,
+    element45: 45,
+    element46: 46,
+    element47: 47,
+    element48: 48,
+    element49: 49,
+    element50: 50,
+    element51: 51,
+    element52: 52,
+    element53: 53,
+    element54: 54,
+    element55: 55,
+    element56: 56,
+    element57: 57,
+    element58: 58,
+    element59: 59,
+    element60: 60,
+    element61: 61,
+    element62: 62,
+    element63: 63,
+    element64: 64,
+    element65: 65,
+    element66: 66,
+    element67: 67,
+    element68: 68,
+    element69: 69,
+    element70: 70,
+    element71: 71,
+    element72: 72,
+    element73: 73,
+    element74: 74,
+    element75: 75,
+    element76: 76,
+    element77: 77,
+    element78: 78,
+    element79: 79,
+    element80: 80,
+    element81: 81,
+    element82: 82,
+    element83: 83,
+    element84: 84,
+    element85: 85,
+    element86: 86,
+    element87: 87,
+    element88: 88,
+    element89: 89,
+    element90: 90,
+    element91: 91,
+    element92: 92,
+    element93: 93,
+    element94: 94,
+    element95: 95,
+    element96: 96,
+    element97: 97,
+    element98: 98,
+    element99: 99
+  )
+}
+
+// This measures the performance of keypath reads though nested structs.
+@inline(never)
+public func run_KeyPathNestedStructs(n: Int) {
+  var sum = 0
+  var index = 0
+  let iterationMultipier = 200
+
+  let singleHopKeyPath0: WritableKeyPath<A, B> = \A.b
+  let singleHopKeyPath1: WritableKeyPath<B, C> = \B.c
+  let singleHopKeyPath2: WritableKeyPath<C, D> = \C.d
+  let singleHopKeyPath3: WritableKeyPath<D, E> = \D.e
+  let singleHopKeyPath4: WritableKeyPath<E, Int> = \E.e
+
+  let appendedKeyPath = singleHopKeyPath0.appending(path: singleHopKeyPath1)
+    .appending(path: singleHopKeyPath2).appending(path: singleHopKeyPath3)
+    .appending(path: singleHopKeyPath4)
+
+  let elementCount = FixedSizeArrayHolder.shared.mainArrayForNestedStructs.count
+  for _ in 1 ... iterationMultipier * n {
+    let element = FixedSizeArrayHolder.shared.mainArrayForNestedStructs[index]
+    sum += element[keyPath: identity(appendedKeyPath)]
```

Comment:
You need to add `identity` outside of the loop, because it adds considerable runtime overhead:
```
let appendedKeyPath = identity(singleHopKeyPath0.appending(path: singleHopKeyPath1)
     .appending(path: singleHopKeyPath2).appending(path: singleHopKeyPath3)
     .appending(path: singleHopKeyPath4))
```

Please also correct this in all other functions.

Reply:
Done!